pandas/core/groupby/generic.py

```diff
@@ -1220,11 +1220,14 @@ def _apply_to_column_groupbys(self, func):
         return func(self)
 
     def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None):
-        """Calculate percent change of each value to previous entry in group"""
-        filled = getattr(self, fill_method)(limit=limit)
-        shifted = filled.shift(periods=periods, freq=freq)
-
-        return (filled / shifted) - 1
+        """Calcuate pct_change of each value to previous entry in group"""
+        with _group_selection_context(self) as new:
+            if fill_method:
+                new = copy.copy(new)
```

Comment:
I continue to need this for state management. On line 835 of groupby/generic.py I see something similar done i.e.
 obj = copy.copy(obj)
 obj._reset_cache()

I know you previously mentioned state management isn't always great, should I take that to mean that it's okay if a call to pct_change changes the state of .obj in groupby? I'll defer to you on this.